server/game/cards/09.4-TCoH/RisingStarsKata.js

```diff
@@ -0,0 +1,53 @@
+import { Durations } from '../../Constants.js';
+const DrawCard = require('../../drawcard.js');
+const AbilityDsl = require('../../abilitydsl');
+const EventRegistrar = require('../../eventregistrar');
+const { CardTypes } = require('../../Constants');
+
+class RisingStarsKata extends DrawCard {
+    setupCardAbilities() {
+        this.duelWinnersThisConflict = [];
+        this.eventRegistrar = new EventRegistrar(this.game, this);
+        this.eventRegistrar.register(['onConflictFinished', 'afterDuel']);
+
+        this.action({
+            title: 'Give a participating unique character +3 military skill',
+            condition: context => context.game.isDuringConflict(),
+            target: {
+                cardType: CardTypes.Character,
+                cardCondition: card => card.isUnique() && card.isParticipating(),
+                gameAction: AbilityDsl.actions.cardLastingEffect(context => ({
+                    duration: Durations.UntilEndOfConflict,
+                    effect: this.duelWinnersThisConflict.includes(context.target) ?
+                        AbilityDsl.effects.modifyMilitarySkill(5) :
+                        AbilityDsl.effects.modifyMilitarySkill(3)
+                }))
+            },
+            effect: 'give {0} +{1} {2} skill until the end of the conflict',
+            effectArgs: context => [this.duelWinnersThisConflict.includes(context.target) ? 5 : 3, 'military'],
+            max: AbilityDsl.limit.perConflict(1)
+        });
+    }
+
+    onConflictFinished() {
+        this.duelWinnersThisConflict = [];
+    }
+
+    afterDuel(event) {
+        if(Array.isArray(event.duel.winner)) {
+            this.duel.winner.forEach(duelWinner => {
+                if(this.duelWinnersThisConflict.indexOf(duelWinner) === -1) {
+                    this.abilities.duelWinnersThisConflict.push(duelWinner);
+                }
+            });
+        }
+        if(event.duel.winner) {
+            this.duelWinnersThisConflict.push(event.duel.winner);
+        }
```

Comment:
Never really looked over this properly, but after fixing Storied defeat, I'm pretty sure there are some bugs here.

You've assigned to `this.abilities.duelWinnersThisConflict` instead of `this.duelWinnersThisConflict`
Also, although not crucial, it should probably be `else if(event.duel.winner)` otherwise you add the array to the duelWinnersThisConflict as well as the individual winners.

Might be worth adding it to the tests?

Reply:
Oh, yeah silly to miss that one.